azure/services/networkinterfaces/spec.go

```diff
@@ -52,6 +53,14 @@ type NICSpec struct {
 	DNSServers                []string
 	AdditionalTags            infrav1.Tags
 	ClusterName               string
+	IPConfigs                 []IPConfig
+}
+
+// IPConfig defines the specification for an IP address configuration.
+type IPConfig struct {
+	PrivateIP       string
+	PublicIP        bool
+	PublicIPAddress string
```

Comment:
Can we instead convert IPConfig struct as below? We will have one less field to manage and a nil pointer will intrinsicly imply weather it is assigned or not?
```suggestion
	PrivateIP       *string
	PublicIPAddress *string
```